compiler/rustc_driver/src/pretty.rs

```diff
@@ -362,7 +362,8 @@ fn write_or_print(out: &str, ofile: Option<&Path>) {
         None => print!("{}", out),
         Some(p) => {
             if let Err(e) = std::fs::write(p, out) {
-                panic!("print-print failed to write {} due to {}", p.display(), e);
+                eprintln!("print-print failed to write {} due to `{}`", p.display(), e);
+                std::process::exit(1);
```

Comment:
Instead of this, what you'd want to do is to pass in `sess: Session` into `write_or_print` and use the diagnostic machinery (`sess.struct_fatal(&format!("pretty-print failed to write {} due to `{e}`", p.display())`).